src/FSharp.Data.GraphQL.Server.AspNetCore/Giraffe/HttpHandlers.fs

```diff
@@ -1,155 +1,268 @@
 namespace FSharp.Data.GraphQL.Server.AspNetCore.Giraffe
 
-open FSharp.Data.GraphQL.Execution
-open FSharp.Data.GraphQL
-open Giraffe
-open FSharp.Data.GraphQL.Server.AspNetCore
-open FsToolkit.ErrorHandling
-open Microsoft.AspNetCore.Http
-open Microsoft.Extensions.DependencyInjection
-open Microsoft.Extensions.Logging
-open System.Collections.Generic
+open System
+open System.IO
 open System.Text.Json
 open System.Text.Json.Serialization
-open System.Threading
 open System.Threading.Tasks
+open Microsoft.AspNetCore.Http
+open Microsoft.Extensions.DependencyInjection
+open Microsoft.Extensions.Logging
+
+open FsToolkit.ErrorHandling
+open Giraffe
+
+open FSharp.Data.GraphQL
+open FSharp.Data.GraphQL.Ast
+open FSharp.Data.GraphQL.Server.AspNetCore
 
 type HttpHandler = HttpFunc -> HttpContext -> HttpFuncResult
 
 module HttpHandlers =
-    open System.Collections.Immutable
-
-    let private httpOk
-        (cancellationToken : CancellationToken)
-        (customHandler : HttpHandler)
-        (serializerOptions : JsonSerializerOptions)
-        payload
-        : HttpHandler =
-        setStatusCode 200
-        >=> customHandler
-        >=> (setHttpHeader "Content-Type" "application/json")
-        >=> (fun _ ctx ->
-            JsonSerializer
-                .SerializeAsync(ctx.Response.Body, payload, options = serializerOptions, cancellationToken = cancellationToken)
-                .ContinueWith (fun _ -> Some ctx) // what about when serialization fails? Maybe it will never at this stage anyway...
-        )
-
-    let private prepareGenericErrors (errorMessages : string list) =
-        (NameValueLookup.ofList [
-            "errors",
-            upcast
-                (errorMessages
-                 |> List.map (fun msg -> NameValueLookup.ofList [ "message", upcast msg ]))
-        ])
-
-    /// HttpHandler for handling GraphQL requests with Giraffe.
-    /// This one is for specifying an interceptor when you need to
-    /// do a custom handling on the response. For example, when you want
-    /// to add custom headers.
-    let handleGraphQLWithResponseInterception<'Root>
-        (cancellationToken : CancellationToken)
-        (logger : ILogger)
-        (interceptor : HttpHandler)
-        (next : HttpFunc)
-        (ctx : HttpContext)
-        =
-        task {
-            let cancellationToken =
-                CancellationTokenSource
-                    .CreateLinkedTokenSource(cancellationToken, ctx.RequestAborted)
-                    .Token
-            if cancellationToken.IsCancellationRequested then
-                return (fun _ -> None) ctx
```

Comment:
I don't know why you removed this logic. This makes sure that aborted requests are closed gracefully.

Reply:
Why is this needed? Isn't using `ctx.RequestAborted` enough?

Reply:
I just use `ctx.RequestAborted` wherever cancellation is possible

Reply:
As I understand reacting to `applicationLifetime.ApplicationStopping` makes sense only for WebSocket connections.
Isn't it? What do you think?

Reply:
It's true that it's much more relevant for WebSocket connections, but here too it's cleaner to take this cancellation token into consideration. Are you concerned that this extra logic is making the code too complex for (almost) no benefit?

Reply:
Fixed, see a new implementation